### PR TITLE
Add CORS support

### PR DESCRIPTION
**Pull request by itajaja**
Hi, I am using caravel as an API service, and CORS would be nice for this use case. I just wanted to know if this looks good, before spending time on tests (which I am not sure how to implement, suggestions welcome!)

### CODE REVIEW

**Comment by itajaja** [comment]
Any reason why this stalled?

File: setup.py [hunk 1]
```diff
@@ -20,6 +20,7 @@
         'cryptography>=1.1.1, <2.0.0',
         'flask-appbuilder>=1.6.0, <2.0.0',
         'Flask-BabelPkg==0.9.6',
+        'Flask-Cors>=2.0.0',
```
Comment (mistercrunch) [inline setup.py]: I would rather not have this as a hard dependency for the vast majority. I'd propose removing it or using `extra_requires`.

File: caravel/__init__.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
 from flask.ext.appbuilder import SQLA, AppBuilder, IndexView
 from flask.ext.appbuilder.baseviews import expose
 from flask.ext.cache import Cache
+from flask.ext.cors import CORS
```
Comment (mistercrunch) [inline caravel/__init__.py]: This should be a conditional import in the if block bellow

**Comment by itajaja** [comment]
@mistercrunch I moved flask-cors into an extra dependency, and I also added some documentation, let me know what you think :)

**Comment by mistercrunch** [comment]
👍

**Comment by itajaja** [comment]
great, thanks

**Pull request merged**